activerecord/lib/active_record/connection_adapters/postgresql/oid/uuid.rb

```diff
@@ -14,7 +14,12 @@ def type
           end
 
           def cast(value)
-            value.to_s[ACCEPTABLE_UUID, 0]
+            casted = value.to_s
+            if casted.match?(ACCEPTABLE_UUID)
```

Comment:
Can you change the `cast` method to private `cast_value` method?
Also, I'd prefer `value if value.match?(ACCEPTABLE_UUID)` style as most other type cast do so.

https://github.com/rails/rails/blob/5ca19efafea08984ff1c464c64969f82b9db3569/activemodel/lib/active_model/type/value.rb#L37-L39

https://github.com/rails/rails/blob/5ca19efafea08984ff1c464c64969f82b9db3569/activemodel/lib/active_model/type/value.rb#L118-L123

Reply:
I intentionally hadn't done the compact style because I was wary of the magical nil return when a conditional is the last statement in a method. But if that's recommended then I don't mind it too badly. I had no idea about the `cast -> cast_value` delegation, so thank you for that information.